tests/bluetooth/bsim_bt/bsim_test_audio/src/ias_client_test.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#ifdef CONFIG_BT_IAS_CLIENT
+
+#include "bluetooth/services/ias.h"
+#include "common.h"
+
+extern enum bst_result_t bst_result;
+
+CREATE_FLAG(g_is_connected);
+CREATE_FLAG(g_service_discovered);
+CREATE_FLAG(g_high_alert_sent);
+CREATE_FLAG(g_mild_alert_sent);
+CREATE_FLAG(g_stop_alert_sent);
+
+static struct bt_conn *g_conn;
+
+static void discover_cb(struct bt_conn *conn, int err)
+{
+	if (err) {
+		FAIL("Failed to discover IAS (err %d)\n", err);
+		return;
+	}
+
+	printk("IAS discovered\n");
+	SET_FLAG(g_service_discovered);
+}
+
+static const struct bt_ias_client_cb ias_client_cb = {
+		.discover = discover_cb,
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	if (err > 0) {
+		char addr[BT_ADDR_LE_STR_LEN];
+
+		bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+		FAIL("Failed to connect to %s (err %u)\n", addr, err);
+		return;
+	}
+
+	g_conn = conn;
+	SET_FLAG(g_is_connected);
+}
+
+static void test_alert_high(struct bt_conn *conn)
+{
+	int err;
+
+	err = bt_ias_client_alert_high(conn);
+	if (err) {
+		SET_FLAG(g_high_alert_sent);
+	}
+}
+
+static void test_alert_mild(struct bt_conn *conn)
+{
+	int err;
+
+	err = bt_ias_client_alert_mild(conn);
+	if (err) {
+		SET_FLAG(g_mild_alert_sent);
+	}
+}
+
+static void test_alert_stop(struct bt_conn *conn)
+{
+	int err;
+
+	err = bt_ias_client_alert_stop(conn);
+	if (err) {
+		SET_FLAG(g_stop_alert_sent);
+	}
```

Comment:
Shouldn't all of these be `!err` or `err == 0`?

Reply:
Done